File: frontend-angular/src/app/models/ingrediente.ts
export type IngredienteBodyRequest = {
  nome: string,
  loja_uuid: string
  produto_uuid: string
  descricao: string
}

export type Ingrediente = {
  nome: string,
  loja_uuid: string
  produto_uuid: string
  descricao: string,
  uuid: string,
}

File: backend/src/infra/database_postgres/entities/ingrediente.py
from src.infra.database_postgres.entities import Base
from sqlalchemy.schema import Column as Col, ForeignKey as FK
from sqlalchemy.types import String as Str, Text


class Ingrediente(Base):   # database
    __tablename__ = "ingredientes"

    uuid = Col(Str(36), primary_key=True, unique=True, nullable=False)
    nome = Col(Text, nullable=False)
    descricao = Col(Text)
    produto_uuid = Col(Str(36), FK("produtos.uuid"), nullable=False)
    loja_uuid = Col(Str(36), FK("lojas.uuid"), nullable=False)
